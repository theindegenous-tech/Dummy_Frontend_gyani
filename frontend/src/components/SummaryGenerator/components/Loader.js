import React from 'react'

const loader = {
  display: 'inline-block',
  verticalAlign: 'text-bottom',
  backgroundColor: 'currentColor',
  borderRadius: '50%',
  width: '0.5rem',
  height: '0.5rem',
}

const Loader = () => (
  <div className='m-4 is-flex is-flex-direction-row is-justify-content-center is-align-content-center'>
    <p className='is-size-3 is-marginless'>Loading</p>
    <div className='p-3'>
      <div style={loader} />
    </div>
    <div className='p-3'>
      <div style={loader} />
    </div>
    <div className='p-3'>
      <div style={loader} />
    </div>
  </div>
)

export default Loader
