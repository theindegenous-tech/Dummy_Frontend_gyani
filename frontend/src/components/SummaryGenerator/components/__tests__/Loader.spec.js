import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Loader from '../Loader'

describe('Loader', () => {
  it('Should render correctly', () => {
    const tree = renderer.create(<Loader />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Should display the "Loading" text', () => {
    const loader = shallow(<Loader />)
    expect(loader.text()).toEqual('Loading')
  })
})
