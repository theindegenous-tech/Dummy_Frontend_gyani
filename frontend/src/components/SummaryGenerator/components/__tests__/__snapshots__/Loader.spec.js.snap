// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Should render correctly 1`] = `
<div
  className="m-4 is-flex is-flex-direction-row is-justify-content-center is-align-content-center"
>
  <p
    className="is-size-3 is-marginless"
  >
    Loading
  </p>
  <div
    className="p-3"
  >
    <div
      style={
        Object {
          "backgroundColor": "currentColor",
          "borderRadius": "50%",
          "display": "inline-block",
          "height": "0.5rem",
          "verticalAlign": "text-bottom",
          "width": "0.5rem",
        }
      }
    />
  </div>
  <div
    className="p-3"
  >
    <div
      style={
        Object {
          "backgroundColor": "currentColor",
          "borderRadius": "50%",
          "display": "inline-block",
          "height": "0.5rem",
          "verticalAlign": "text-bottom",
          "width": "0.5rem",
        }
      }
    />
  </div>
  <div
    className="p-3"
  >
    <div
      style={
        Object {
          "backgroundColor": "currentColor",
          "borderRadius": "50%",
          "display": "inline-block",
          "height": "0.5rem",
          "verticalAlign": "text-bottom",
          "width": "0.5rem",
        }
      }
    />
  </div>
</div>
`;
